Automattic/jetpack
# My Jetpack: Add import detection codesniffs (PHPCS)

Add detection unimported Symbols (eventual Fatals) as refrences to global class names from namespaced files

#### Changes proposed in this Pull Request:

* Adds [phpcs-import-detection](https://github.com/sirbrillig/phpcs-import-detection) as dependency of codesniffer
* Sets rules in My Jetpack's `.phpcs.xml` to raise errors for unimported symbols

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:
1. Checkout this branch
2. run `jetpack install -all`
3. Update `projects/packages/my-jetpack/src/products/class-backup.php` to remove a Line that reads `use Jetpack`.
5. Run `composer phpcs:lint projects/packages/my-jetpack/src/products/`
  ![image](https://user-images.githubusercontent.com/746152/155696347-6d186f42-d9bb-4af0-8c86-0311ea3947cf.png)
7. Confirm you see a report about `Jetpack` not being imported.
8. Run `git commit -m 'Trying out' projects/packages/my-jetpack/src/products/class-backup.php`
9. Confirm you get blocked by the linter
  <img width="1244" alt="image" src="https://user-images.githubusercontent.com/746152/155696499-46a6b190-d34d-4356-b67a-1b7916ff2d29.png">
